packages/applications-service-api/src/repositories/representation.backoffice.repository.js

```diff
@@ -0,0 +1,13 @@
+const { prismaClient } = require('../lib/prisma');
+
+const getRepresentationById = async (representationId) => {
+	return prismaClient.representation.findUnique({
+		where: {
+			representationId
+		}
```

Comment:
We could fetch the `represented` and `representative` details here as part of this query:
```
{
  include: {
    represented: true,
    representative: true
  }
}
```

We would need to add a relation in the schema:
```
representative         ServiceUser?               @relation("representative", fields: [representativeId], references: [serviceUserId], onDelete: NoAction, onUpdate: NoAction)
  represented            ServiceUser?               @relation("represented", fields: [representedId], references: [serviceUserId], onDelete: NoAction, onUpdate: NoAction)
```

It does end up doing separate queries rather than JOINs so the benefit is more about leveraging Prisma rather than implementing the extra service user repository to do the same thing

Reply:
Yeah we considered having an actual relation between them in the schema but we can't guarantee what order the messages will come in from BO so if get a represented (service user) first but would not have the representation. I believe @Philip-Philippou-KC also mentioned another issue with having a relation in prisma on this.

Reply:
The ordering is the big one, we need to make message processing idempotent, if there is no service user there we will be constraints errors.

Reply:
As discussed - foreign key constraints prevent this from being feasible due to the `Representation` row depending on the `ServiceUser` row being inserted first, which we can't guarantee due to messages from the queue possibly arriving out of order.